### PR TITLE
Add files for packaging into deb

### PR DESCRIPTION
**Pull request by hosiet**
Hi all,

I came back with a PR for pi-blaster. This one still has nothing to do with C source code (as I found it rather complicated to fully understand), but provides files essential to make pi-blaster into deb package. This may be useful, because `make install` and `make uninstall` is not friendly to package management system at all.

A full list of changes is written below:
* Modify `Makefile.am` and `configure.ac` to provide a new configure-time option to ignore the system-service-installation process. This option is used in deb packaging.
* Add all necessary files into `debian` dir.
* Move and rename the original sysvinit / systemd service files into `debian` dir. Symlinks are left for compatibility with original Makefile.
* Modify `README.md` for new installation / uninstallation instructions.

Besides, I suggested a minor version bump in the PR since a lot of changes have been introduced since the last update of the version number. If you think it really unnecessary, I will cancel the version number change inside the PR.

I am sure that there must be some kind of mistakes or something similar in the PR. Please review it and give your opinion. Thanks.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,6 +28,34 @@ To start pi-blaster and have it relaunched automatically on every reboot:
...
+#### Raspbian 8 ("Jessie") users
+
+If you have upgraded your Raspberry Pi to use Raspbian 8 with systemd, you may
+still follow the instruction given for Wheezy users. However, an extra
+special step may help you to better integrate with systemd.
+
+    sudo apt-get install debhelper dh-autoreconf dh-systemd dpkg-dev \
+         init-system-helpers autoconf
+    cd /path/to/pi-blaster/
+    git revert 656e14c6acbb77aec95a57783726c043ee642f34
```
Comment (sarfata) [inline README.md]: Why is this needed? This is a pretty convoluted way of switching between with and without systemd. Maybe an easier way would be to just duplicate the debian files in debian7 and debian8 and symlink before building? 
Does Debian provide a standard way to handle this?

I am worried that relying on a git revert is going to seriously confuse people.

File: README.md [hunk 2]
```diff
@@ -140,12 +174,12 @@ This library was developed for TBideas high power LED driver. You can read more
 
 ## Contributors
 
-Pete Nelson (https://github.com/petiepooo)
-Edgar Siva (https://github.com/edgarsilva)
-Alex Lennon (https://github.com/ajlennon)
-Lara Maia (https://github.com/LaraCraft304)
-Pattrick Hüper (https://github.com/phueper)
-Boyuan Yang (https://github.com/hosiet)
+* Pete Nelson (https://github.com/petiepooo)
```
Comment (sarfata) [inline README.md]: :+1:

File: Makefile.am [hunk 3]
```diff
@@ -7,6 +7,7 @@ EXTRA_DIST = pi-blaster.service
 sbin_PROGRAMS = pi-blaster
 pi_blaster_SOURCES = pi-blaster.c mailbox.c
 
+if !DEBIAN_PKG
```
Comment (sarfata) [inline Makefile.am]: Can you also change the the init_d_SCRIPTS (you moved the file to the debian subfolder).

**Comment by sarfata** [comment]
Great pull-request. So sorry it took me so long to look at it. Please make the few changes I suggested and I will merge it.

File: README.md (hunk 1 above)
Comment (hosiet) [inline README.md]: This is a difficult problem: some people are still using Wheezy (which is `oldstable`, because default Raspbian wasn't upgraded to Jessie until September 2015), and Wheezy has neither `dh-systemd` nor `init-system-helpers` package. If the default debian rule is written to add systemd support (which is Debian's standard solution to the coexistence between systemd and sysvinit on Jessie), those users will need to **manually** enable `debian-backports` for those two packages that was backported to Wheezy even though they are not really using systemd. Maybe it's not a good idea either.

Using `git revert` is indeed not very good-looking. Will a wrapper script (e.g. `autobuild`) and symlink be better?

File: Makefile.am (hunk 3 above)
Comment (hosiet) [inline Makefile.am]: Changing `init_d_SCRIPTS` looks pretty strange. Besides backward-incompatibility, the init scripts in `/etc/init.d` should be anything like `foobar` or `foobar.sh` but `pi-blaster.init`. The file named `foo-bar.init` is only used when packaging. It looks OK if only installed manually from source.

And if the name really should be changed, it would be great if there are some solid suggestions (like removing current symlink, renaming the symlink or making a duplication of init script, etc).

Comment (sarfata) [inline Makefile.am]: You have removed the file `pi-blaster.boot.sh` so it will not work anymore. We need some way to keep it working for people who are not using systemd and not using the debian package.

This [StackOverflow question](http://stackoverflow.com/questions/26837148/autoconf-and-init-scripts) offers two solutions. The first one is very debian specific and you seem to be more familiar with dh-tools than I am. 

The second one suggests how to cp the file in an `install-exec-hook`. We could use that.

File: README.md (hunk 1 above)
Comment (sarfata) [inline README.md]: Wrapper script sounds good to me.

**Comment by hosiet** [comment]
I tried to figure out the problem recently, and I would like to make things more clear here:

## Target
* Allow user to package and install using `deb` file
* Keep the original behaviour (which is, startup immediately after installation finished, auto startup on system boot), regardless of the installation method (build and install from source code directly, install via deb package, etc)

## Current problems
+  #57 have shown that after installed directly from source on a system using systemd, `pi-blaster` won't start automatically after installation and on system boot, but needs an extra step of `systemctl enable pi-blaster && systemctl start pi-blaster`.
    * This can be fixed by adding corresponding `install-data-hook` and `uninstall-hook` section in `Makefile.am`, just as the existing example in `Makefile.am` *before*, which is used to work with init scripts. Example diff could be as below, and I believe that it could solve the problem, though I haven't tested it yet:
```
diff --git a/Makefile.am b/Makefile.am
index 9820d2b..c623c9e 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -10,7 +10,16 @@ pi_blaster_SOURCES = pi-blaster.c mailbox.c
 if !DEBIAN_PKG
 if HAVE_SYSTEMD
 systemdsystemunit_DATA = pi-blaster.service
-else
+if !CROSS_COMPILING
+install-data-hook:
+       systemctl enable pi-blaster
+       systemctl start pi-blaster
+
+uninstall-hook:
+       systemctl stop pi-blaster
+       systemctl disable pi-blaster
+endif # !CROSS_COMPILING
+else # !HAVE_SYSTEMD
 init_ddir = $(sysconfdir)/init.d
 init_d_SCRIPTS = pi-blaster.boot.sh
```
+  Building deb package on Wheezy with systemd-related stuff requires tools from `debian-backports`, which is unfriendly to end-users. But throwing systemd stuff away may not be a good idea, because native systemd service may works better. Here are some choices:
    * Completely throw away systemd service in `deb` packaging. Installation using deb package would still use init scripts, like in good old days.
    * Pack the `deb` using modern methods, and leave the installation of dependencies like [init-system-helpers](https://packages.debian.org/wheezy-backports/init-system-helpers) and [dh-systemd](https://packages.debian.org/wheezy-backports/dh-systemd) to users. This means easy work to upstream because we only need to provide single version of files in `/debian/` dir. Only those using Wheezy would be affected, and people using Jessie would be happy and unaffected at all.
    * Write a wrapper script (i.e. `autobuild`) and modify files in `/debian/` dir on demand according to current debian version. This means a little more work to do.

## Solutions
* The solution to the first problem has been proposed.
* The solution to the second problem seems to be wrapper scripts.

**Comment by sarfata** [comment]
@hosiet Sorry it took me so long to get back to you.

I am going to merge this PR as-is. It is much better than everything else we have right now but I am going to revert the commit that removed systemd. I will put in the readme that the debian package is recommended only for Jessie user for the moment. I hope that is ok with you.

Feel free to let me know here - or in another PR if you have a better way to do this.

**Comment by hosiet** [comment]
That's fine - the dicision is up to you. Actually I am a little busy
recently, but I will take a review later and see what is going on recently
(e.g., rpi3 support). Feel free to contact me if there are any related bugs.

**Comment by sarfata** [comment]
Closed by #64. Thanks a lot @hosiet
